File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        if(!word.contains(Character.toString(ch))) return word;
        char[] newWord = word.toCharArray();
        int j = word.indexOf(ch);
        int i = 0;
        while(i<j){
            char temp = newWord[i];
            newWord[i] = newWord[j];
            newWord[j] = temp;
            i++;
            j--;
        }
        return new String(newWord);
    }
}

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function(word, ch) {
    let arr = []
    for(let s of word){
        if(s !== ch){
            arr.push(s)
        }else {
            arr.push(s)
            return arr.reverse().join("") + word.substring(word.indexOf(s) + 1)
        }
    }
    return word
};